老國音.js

```diff
@@ -0,0 +1,292 @@
+  ['幽韻', [['幫滂並母', 'iau'], ['', 'iəu']]],
+
+  //咸攝
+  ['鹽添韻 或 咸銜嚴韻 牙喉音', 'ian'],
+  ['覃談咸銜嚴韻', 'an'],
+  ['凡韻', 'yan'],
+
+  //深攝
+  ['侵韻', 'in'],
+
+  //山攝
+  ['寒刪山韻 合口', 'uan'],
+  ['刪山韻 牙喉音', 'ian'],
+  ['寒刪山韻', 'an'],
+  ['仙先韻', [['合口', 'yan'], ['', 'ian']]],
+  ['元韻', [['開口', 'ian'], ['', 'yan']]],
+
+  //臻攝
+  ['痕韻', 'ən'],
+  ['魂韻', 'uən'],
+  ['真臻殷韻', [['合口', 'yn'], ['', 'in']]], //臻韻莊組會推如ən，詳後
+  ['文韻', 'yn'],
+
+  //梗攝
+  ['梗攝 二等', [['合口', 'uŋ'], ['', 'əŋ']]],
+  ['梗攝', [['合口', 'yŋ'], ['', 'iŋ']]],
+
+  //曾攝
+  ['登韻', [['合口', 'uŋ'], ['', 'əŋ']]],
+  ['蒸韻', [['合口', 'yŋ'], ['', 'iŋ']]], //蒸合韻無舒聲但照推
+
+  //宕攝
+  ['唐韻', [['合口', 'uaŋ'], ['', 'aŋ']]],
+  ['陽韻', [['合口 或 幫莊組', 'uaŋ'], ['', 'iaŋ']]],
+
+  //江攝
+  ['江韻', [['牙喉音', 'iaŋ'], ['知莊組', 'uaŋ'], ['', 'aŋ']]],
+
+  //通攝
+  ['通攝', [['三等 牙喉音', 'yŋ'], ['', 'uŋ']]],
+], '無韻母規則');
+
+const 入聲韻母規則 = () => when([
+  //咸攝
+  ['覃談韻', [['牙喉音', 'oʔ'], ['', 'aʔ']]],
+  ['咸銜韻', [['牙喉音', 'iaʔ'], ['', 'aʔ']]],
+  ['嚴鹽添韻', 'ieʔ'],
+  ['凡韻', [['脣音', 'uaʔ'], ['', 'yeʔ']]], //明凡入無字
+
+  //深攝
+  ['侵韻', [['莊組', 'əʔ'], ['', 'iʔ']]],
+
+  //山攝
+  ['寒韻', [['合口', 'uoʔ'], ['脣牙喉音', 'oʔ'], ['', 'aʔ']]],
+  ['刪山韻', [['合口', 'uaʔ'], ['牙喉音', 'iaʔ'], ['', 'aʔ']]],
+  ['仙先韻', [['合口', 'yeʔ'], ['', 'ieʔ']]],
+  ['元韻', [['脣音', 'uaʔ'], ['合口', 'yeʔ'], ['', 'ieʔ']]],
+
+  //臻攝
+  ['痕韻', 'əʔ'],
+  ['魂韻', [['脣音', 'oʔ'], ['', 'uʔ']]],
+  ['真韻 合口', [['知莊組', 'uoʔ'], ['', 'yʔ']]], //知莊組uʔ、uoʔ兩者勢均，徹莊母多uʔ
+  ['真臻殷韻', 'iʔ'], //莊組臻韻難決定取əʔ還是ʅʔ
+  ['文韻', 'yʔ'], //見組文韻入聲讀音預測yʔ（如「屈ㄑㄩ˙」，溪文入）但見羣疑母多yeʔ
+
+  //梗攝
+  ['梗攝 二等', [['合口', 'uoʔ'], ['', 'əʔ']]],
+  ['梗攝', [['合口', 'yʔ'], ['', 'iʔ']]],
+
+  //曾攝
+  ['登韻', [['合口', 'uoʔ'], ['', 'əʔ']]],
+  ['蒸韻', [['合口', 'yʔ'], ['莊組', 'əʔ'], ['', 'iʔ']]],
+
+  //宕攝
+  ['唐韻', [['合口', 'uoʔ'], ['', 'oʔ']]],
+  ['陽韻', [['合口 或 幫莊組', 'yoʔ'], ['', 'ioʔ']]], //明陽入無字；孃陽入無常用字；莊組開口oʔ、uoʔ兩者勢均，但對應陽聲韻今讀合口呼
+
+  //江攝
+  ['江韻', [['牙喉音', 'ioʔ'], ['知莊組', 'uoʔ'], ['', 'oʔ']]],
+
+  //通攝
+  ['通攝', [['三等 牙喉音', 'yʔ'], ['', 'uʔ']]],
+], '無韻母規則');
+
+const 聲調規則 = () => when([
+  ['平聲', [['清音', '1'], ['濁音', '2']]],
+  ['去聲 或 上聲 全濁', '4'],
+  ['上聲', '3'],
+  ['入聲', '5'],
+], '無聲調規則');
+
+let 聲母 = 聲母規則();
+let 韻母 = is`舒聲` ? 舒聲韻母規則() : 入聲韻母規則();
+let 聲調 = 聲調規則();
+
+if (選項.常母平聲陰聲韻聲母和船母平聲聲母 && is`(常母 陰聲韻 或 船母) 平聲`) 聲母 = 選項.常母平聲陰聲韻聲母和船母平聲聲母;
+if (選項.見泰合 && is`見溪羣母 泰韻 合口`) 韻母 = 選項.見泰合;
+if (選項.見佳合 && is`見影匣母 佳韻 合口`) 韻母 = 選項.見佳合;
+if (選項.幫幽 && is`幫滂並母 幽韻`) 韻母 = 選項.幫幽;
+if (選項.莊侵入 && is`莊組 侵韻 入聲`) 韻母 = 選項.莊侵入;
+if (選項.知莊真合入 && is`知莊組 真韻 合口 入聲`) 韻母 = 選項.知莊真合入;
+if (選項.莊臻入 && is`莊組 臻韻 入聲`) 韻母 = 選項.莊臻入;
+if (選項.見文入 && is`見羣疑母 文韻 入聲`) 韻母 = 選項.見文入;
+if (選項.見梗二開舒 && is`溪疑影曉匣母 梗攝 二等 開口 舒聲`) 韻母 = 選項.見梗二開舒;
+if (選項.匣上 === false && is`匣母 上聲`) 聲調 = '3';
+
+//顎化；疑母齊、撮呼不規則脫鼻
+if (['i', 'y'].includes(韻母[0])) 聲母 = {
+  k: 'tɕ', kʰ: 'tɕʰ', x: 'ɕ',
+}[聲母] || 聲母;
+if (['i', 'y'].includes(韻母[0]) && 聲母 === 'n' && !is`齊之韻 開口`) 聲母 = 'ɲ';
+if (韻母[0] === 'i' && 聲母 === 'ŋ') {
+  if (is`(齊之尤蒸陽韻 非 合口) 或 (侵仙先元庚韻 入聲)`) 聲母 = 'ɲ'; //要不要括著？
+  else 聲母 = '';
+}
```

Comment:
```suggestion
if (韻母[0] === 'i' && 聲母 === 'ŋ') 聲母 = is`(齊之尤蒸陽韻 非 合口) 或 (侵仙先元庚韻 入聲)` ? 'ɲ' : '';
```
至於要不要括住，我沒有意見，看 @syimyuzya